src/lib.rs

```diff
@@ -456,6 +536,7 @@ pub fn run_fixing_for_slice<'a>(
     if pending_fixes.is_empty() {
         return violations;
     }
+    drop(from_file_run_context_instance_provider);
```

Comment:
Good job Rust, you pointed out something that I had not thought about at all (in trying to think through the "incremental provided-type re-computation" stuff) - if the provided-type instances have references into the source text and then we send that source text off to the "fixing loop" to get mutated (and we're still "holding onto" the provided-type instances), that's bad!

So ya haven't tried to think through that yet, likely might make sense to just not allow provided types to store references into the source text (but it's a trade-off: that would mean slightly slower "single run" I'd think (because you'd have to allocate (or maybe "small-string"?) the bits of source text you wanted to store) but if there were actually incremental provided-type re-computation for some things then probably some speedup from that each time through the fixing loop)?

Reply:
Mmm, right.